features/FEATURE_BLE/ble/BLE.h

```diff
@@ -195,127 +380,103 @@ class BLE
-     * interrupt, but the MCU might wake up several times before
-     * returning (to service the stack). This is not always interchangeable with
-     * WFE().
+    /*
+     * Deprecation alert!
+     * All of the following are deprecated and may be dropped in a future
+     * release. Documentation should refer to alternative APIs.
      */
-    void waitForEvent(void);
-
 public:
     /**
-     * The value of the BLE::InstanceID_t for the default BLE instance.
-     */
-    static const InstanceID_t DEFAULT_INSTANCE = 0;
-#ifndef YOTTA_CFG_BLE_INSTANCES_COUNT
-    /**
-     * The number of permitted BLE instances for the application.
-     */
-    static const InstanceID_t NUM_INSTANCES = 1;
-#else
-    /**
-     * The number of permitted BLE instances for the application.
-     */
-    static const InstanceID_t NUM_INSTANCES = YOTTA_CFG_BLE_INSTANCES_COUNT;
-#endif
-
-    /**
-     * Get a reference to the BLE singleton corresponding to a given interface.
-     * There is a static array of BLE singletons.
-     *
-     * @note Calling Instance() is preferred over constructing a BLE object
-     * directly, as it returns references to singletons.
-     *
-     * @param[in] id
-     *              Instance-ID. This should be less than NUM_INSTANCES
-     *              for the returned BLE singleton to be useful.
-     *
-     * @return A reference to a single object.
-     */
-    static BLE &Instance(InstanceID_t id = DEFAULT_INSTANCE);
-
-    /**
      * Constructor for a handle to a BLE instance (the BLE stack). BLE handles
      * are thin wrappers around a transport object (that is, ptr. to
      * BLEInstanceBase).
      *
      * It is better to create BLE objects as singletons accessed through the
      * Instance() method. If multiple BLE handles are constructed for the same
-     * interface (using this constructor), they will share the same underlying
+     * interface (using this constructor), they share the same underlying
      * transport object.
+     *
+     * @deprecated Use the Instance() function instead of the constructor.
      */
+    MBED_DEPRECATED("Use BLE::Instance() instead of BLE constructor.")
     BLE(InstanceID_t instanceID = DEFAULT_INSTANCE);
 
     /**
-     * Fetch the ID of a BLE instance. Typically there would only be the DEFAULT_INSTANCE.
-     */
-    InstanceID_t getInstanceID(void) const {
-        return instanceID;
-    }
-
-    /*
-     * Deprecation alert!
-     * All of the following are deprecated and may be dropped in a future
-     * release. Documentation should refer to alternative APIs.
+     * Yield control to the BLE stack or to other tasks waiting for events. This
+     * is a sleep function that returns when there is an application-specific
+     * interrupt, but the MCU might wake up several times before returning (to
+     * service the stack). This is not always interchangeable with WFE().
+     *
+     * @deprecated This function block the CPU prefer to use the pair
```

Comment:
Query: Is this sentence saying that this function blocks the CPU, or that this is a function block that the CPU prefers to use or something else?

Reply:
I suppose this can be rephrase as follows: 

> This is a sleep function that returns when there is an application-specific interrupt. This is not interchangeable with WFE() considering that the MCU might wake up several times to service the stack before returning control to the caller.